### PR TITLE
[pfedit] Remove `start_proof` stub from `Pfedit`

### PR DESCRIPTION
**Pull request by ejgallego**
This way we only have 2 `start_proof` entries, in `Lemmas` and
`Proof_global`; which they should be unified / brought closer in the
future.

### CODE REVIEW

**Comment by ejgallego** [comment]
Should be ready. Overlay:
- https://github.com/coq-community/paramcoq/pull/15

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
This needs a rebase now it seems.

**Comment by ejgallego** [comment]
Done thanks.

**Comment by ppedrot** [comment]
Something went wrong and I don't know how much is unrelated to this PR. @ejgallego can you rebase again?

**Comment by ejgallego** [comment]
> Something went wrong and I don't know how much is unrelated to this PR. @ejgallego can you rebase again?

I will, but what's wrong here is not merged overlays in #8902 . I'll wait for the overlays to be merged then relaunch.

Note that in #9022 I proposed a new team "overlay-maintainers" to alleviate this issue, so we can directly merge the corresponding overlays.

Generally, each day I like monorepos more, but that's a different discussion IMO.

**Comment by ejgallego** [comment]
Green now.

**Comment by ejgallego** [comment]
Merci

**Pull request merged**
